### PR TITLE
Fix fees for trampoline relay

### PR DESCRIPTION
**Pull request by thomash-acinq**
When using multipart payments, the path-finding request sent to the router does not match the actual payment, it is only a request to route a fraction of the amount but the fee budget is still the one for the full payment. As a consequence the router will return routes that will not satisfy the fee budget.
The fee budget is checked one last time before relaying the payment but ignoring the fee of the first channel (which should not be ignored for trampoline relays). I fix this last check.

### CODE REVIEW

**Review by pm47 — COMMENTED** [review]
This looks good to me, but I'm not familiar enough with that part of the code to ack. Let's wait for a review from @t-bast.

**Review by t-bast — COMMENTED** [review]
Nice catch, looks good to me, only a few nits

File: eclair-core/src/test/scala/fr/acinq/eclair/router/RouteCalculationSpec.scala
Comment (thomash-acinq) [reply eclair-core/src/test/scala/fr/acinq/eclair/router/RouteCalculationSpec.scala]: Done, I put these values originally because they were what we had in the case I was trying to reproduce.

**Review by t-bast — APPROVED** [review]

**Review by t-bast — APPROVED** [review]

**Pull request merged**
